com.revolutionarygamesstudio.ThriveLauncher.yaml

```diff
@@ -0,0 +1,62 @@
+app-id: com.revolutionarygamesstudio.ThriveLauncher
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+command: thrivelauncher
+separate-locales: false
+rename-icon: thrive-launcher
+finish-args:
+  - --share=network
+  - --device=dri
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=ipc
+modules:
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: thrive-launcher
+    buildsystem: simple
+    cleanup:
+      - squashfs-root
+    sources:
+      - type: file
+        url: https://github.com/Revolutionary-Games/Thrive-Launcher/releases/download/v1.3.1/Thrive-Launcher-1.3.1.AppImage 
+        sha256: b63529154c5e7f29f2a92c206b692fa3de1b50364a9689f8a44f2eb5845b53e1
+        x-checker-data:
+          type: git
```

Comment:
Have you verified it works? You're not using `type: git` here to use the git checker.

Reply:
I have 0 clue what you are reading.
`type: git` would not work because this is *not* using a git repository, it is downloading an AppImage

Reply:
And yet, you are using `git` for the checker. You cannot not use git as the source and use git checker at the same time. So to repeat my question, have you actually tested it?

Reply:
I ran ```flatpak run org.flathub.flatpak-external-data-checker ``` without any issues

Reply:
…which means you haven't actually tested it detects new releases, have you? If you change the version to 1.3.0, it won't detect the new release at all because it's not a git module. At the very least, you need this:

```
diff --git a/com.revolutionarygamesstudio.ThriveLauncher.yaml b/com.revolutionarygamesstudio.ThriveLauncher.yaml
index 0be11ed..dcf73ed 100644
--- a/com.revolutionarygamesstudio.ThriveLauncher.yaml
+++ b/com.revolutionarygamesstudio.ThriveLauncher.yaml
@@ -32,8 +32,10 @@ modules:
         url: https://github.com/Revolutionary-Games/Thrive-Launcher/releases/download/v1.3.1/Thrive-Launcher-1.3.1.AppImage 
         sha256: b63529154c5e7f29f2a92c206b692fa3de1b50364a9689f8a44f2eb5845b53e1
         x-checker-data:
-          type: git
-          version-scheme: semantic  
+          type: json
+          url: https://api.github.com/repos/Revolutionary-Games/Thrive-Launcher/releases/latest
+          version-query: ".tag_name | sub(\"^v\"; \"\")"
+          url-query: ".assets[] | select(.name==\"Thrive-Launcher-\" + $version + \".AppImage\") | .browser_download_url"
 
       - type: script
         dest-filename: thrivelauncher
```

And then you also need to replace `1.3.1` with regexp of wildcard in `build-commands` as well, as it will fail to build the next update.

On that note, I do assure you that I have generally better things to do than having these fruitless discussions that could have been resolved much faster if you were more cooperative. Let's respect each other time – if I mention something, it's not because I'm frustrated and want to nitpick on your submissions, but because it's most likely a valid concern that something could have been done better.

Reply:
I apologize for being frustrating in this regard. I was honestly confused and my own stubbornness gets in the way of me doing things at times.

I have merged your suggestion, It does in fact work well.